server/src/main/java/org/apache/druid/client/cache/ForegroundCachePopulator.java

```diff
@@ -76,20 +75,18 @@ public ForegroundCachePopulator(
             sequence,
             input -> {
               if (!tooBig.get()) {
```

Comment:
Please justify the use of `AtomicBoolean` in a comment, or use a `MutableBoolean` if concurrency is not required.

Reply:
Oops good catch, yeah this is probably not required, will change :+1: